### PR TITLE
Added support for network retry

### PR DESCRIPTION
**Pull request by kensi**
Signed-off-by: Kenny Stridh <kenny.stridh@gmail.com>

Fixes #?

### Context
Better support for unreliable network

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by donat** [comment]
@gradle/execution This PR makes sense to me, but I'd like you to make the final call.

**Comment by donat** [comment]
@bot-gradle test rFM

File: subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientHelper.java [hunk 1]
```diff
@@ -222,6 +223,7 @@ private synchronized CloseableHttpClient getClient() {
         if (client == null) {
             HttpClientBuilder builder = HttpClientBuilder.create();
             new HttpClientConfigurer(settings).configure(builder);
+            builder.setRetryHandler(new DefaultHttpRequestRetryHandler(3, true));
```
Comment (kensi) [inline subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientHelper.java:226]: I think we could skip this part, haven't seen any major improvement in fetch operations after this change.

**Comment by kensi** [comment]
Sorry, been stuck in escalations the last week. 

I think that the important part is the retry for publish, should be safe as well. I can remove the generic retry for now and rework it to be either configurable or only applicable for fetching artifacts for example

**Pull request closed without merging**